subprojects/language-java/src/test/groovy/org/gradle/api/internal/tasks/compile/processing/IncrementalFilerTest.groovy

```diff
@@ -40,12 +40,13 @@ abstract class IncrementalFilerTest extends Specification {
 
     abstract Filer createFiler(Filer filer, AnnotationProcessingResult result, Messager messager)
 
-    def "fails when a package element is given as an originating element"() {
+    def "packages are valid originating elements"() {
```

Comment:
Might be worth checking what happens with `module-info` files too. I'm not sure what annotation processors allow in this case.